aip/general/0236.md

```diff
@@ -0,0 +1,297 @@
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of the preview.
+  google.protobuf.Timestamp stop_time = 4;
+}
+```
+- PreviewMetadata **must** have the fields defined in the proto above and **must** use the same set of values for state.
+- A shared proto used across services **must not** be used ([AIP-213][aip-123]).
+- When a preview is first started, PreviewMetadata **must** be empty. It will only be present on experiments once a custom verb has been used to start it.
+- All fields **must** be output only.
+- The state field **must** only be updated by the Control Plane:
+  - The state **should** subsequently be altered between ACTIVE and INACTIVE to start or stop downstream evaluation using custom verbs described below.
+- The first time the startPreview custom verb is used, the Control Plane **must** create the PreviewMetadata and adds: 
+  - state = ACTIVE
+  - log_type string which is common to all preview mode logs for the product.
+  - start_time
+- When the stopPreview custom verb is used, the Control Plane **must** update PreviewMetadata with
+  - state = INACTIVE
+  - stop_time
+- start_time **must** be recreated every time the status is changed to ACTIVE.
+
+[aip-123]: https://aip.dev/123
+
+### Extensibility to Collection use-case
+
+For the Collection use-case, Rule is a repeated nested message on a Policy resource as shown below.
+
+```proto
+message Policy {
+
+  string name = 1;
+
+  Rule[] rules = 2;
+
+  // Other existing fields…
+}
+```
+
+A experiment is created for the entire parent resource. So essentially, there will be a experiment with its own set of rules. The following guidelines **should** be used to determine how to author an experiment when “rules” is a repeated nested message.
+
+To show the deltas between a live resource and its experiment, the following example will be used:
+A live resource has rules: [rule1, rule2]
+
+Each example below will show the value of the spec field in the experiment, and the intended result.
+
+Create rule3 in preview mode:
+Experiment resource rules: [rule1, rule2, rule3]
+
+Preview the deletion of rule 2:
+Experiment resource rules: [rule1]
+
+Update rule2 to rule2’:
+Experiment resource rules: [rule1, rule2’]
+
+### APIs
+
+Experiments **must** follow resource-oriented design as stated in ([AIP-121][aip-121]). Additionally, the following guidance **must** be followed which is specific to Experiments.
+
+[aip-121]: https://google.aip.dev/121#methods
+
+#### create
+
+- Creating a new Experiment for preview mode **should** support the following use-cases:
+  - Preview a new Policy.
+  - Updating a current Policy.
+  - Preview the deletion of a current Policy.
+- Previewing a change to a current Policy (update or delete use-cases) **should** require copying the full payload of the live resource into the inlined Policy field and making changes to fields-under-test as needed for preview. Delete requires setting rules collection to null.
```

Comment:
Should this be "or"? "copying the full payload of the resource into the inlined Policy field or making changes to fields-under-test..."?

"field-under-test" should not be hyphenated. Just say, "the fields under test".

The sentence starting, "Previewing a change..." is cumbersome and hard to understand. Please see if you can change from passive to active voice and consider breaking into two sentences.

Reply:
This comment was missed.